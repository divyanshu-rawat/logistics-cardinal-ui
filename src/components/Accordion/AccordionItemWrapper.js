import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Fade from 'react-reveal/Fade';
import AccordionItemHeader from './AccordionItemHeader';
import AccordionItemContent from './AccordionItemContent';

export const AccordionItemDividerStyled = styled('div')`
  border-bottom: 2px solid #2c3037;
`;
class AccordionItemWrapper extends PureComponent {
  static propTypes = {
    registerComponent: PropTypes.func.isRequired,
    id: PropTypes.string.isRequired,
    items: PropTypes.object.isRequired,
    content: PropTypes.node.isRequired,
    title: PropTypes.node.isRequired,
    toggleItem: PropTypes.func.isRequired,
    expandOnInit: PropTypes.bool.isRequired,
  };

  static ANIMATION_DELAY = 400;
  static ANIMATION_DURATION = 700;

  componentDidMount() {
    const { registerComponent, id, expandOnInit } = this.props;
    registerComponent(id, expandOnInit);
  }

  render() {
    const { items, toggleItem, id, title, content } = this.props;
    return (
      <AccordionItemDividerStyled>
        <AccordionItemHeader action={toggleItem(id)} open={!!items[id]}>
          {title}
        </AccordionItemHeader>
        <Fade
          delay={AccordionItemWrapper.ANIMATION_DELAY}
          duration={AccordionItemWrapper.ANIMATION_DURATION}
          collapse
          when={!!items[id]}
        >
          <AccordionItemContent>{content}</AccordionItemContent>
        </Fade>
      </AccordionItemDividerStyled>
    );
  }
}

export default AccordionItemWrapper;
