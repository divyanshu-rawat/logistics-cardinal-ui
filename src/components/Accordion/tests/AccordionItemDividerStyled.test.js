import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { AccordionItemDividerStyled } from '../AccordionItemWrapper';

const renderedComponent = () => shallow(<AccordionItemDividerStyled />);

describe('AccordionItemDividerStyled', () => {
  it('should render the component', () => {
    const renderComponent = renderedComponent();
    expect(toJson(renderComponent)).toMatchSnapshot();
  });
});
