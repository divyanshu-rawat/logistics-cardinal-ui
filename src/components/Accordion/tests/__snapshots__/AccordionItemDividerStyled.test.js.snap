// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionItemDividerStyled should render the component 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "AccordionItemWrapper__AccordionItemDividerStyled-eXMyLT",
        "isStatic": false,
        "rules": Array [
          "border-bottom:2px solid ",
          [Function],
          ";",
        ],
      },
      "displayName": "AccordionItemWrapper__AccordionItemDividerStyled",
      "render": [Function],
      "styledComponentId": "AccordionItemWrapper__AccordionItemDividerStyled-eXMyLT",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
